### DISCUSSION TITLE
Connecting to Microsoft Surface

### DISCUSSION THREAD

**Discussion by 3PY44** [artisan-roaster-scope/artisan#1220] [Q&A]
I have a Microsoft Surface Pro 9, and a Hottop 2k+ roaster. When attempting to use Artisan I cannot get my Surface to read the machine being plugged in. 
When I configure Artisan for the Hottop, I can go through all the steps up to choosing the port I want. Artisan only shows 1 port being available to use, but it doesn't actually work. I cannot get it to recognize it is plugged into my computer. 
Fixes I have tried: Deleting and reinstalling Artisan (newest version), getting a new cord to plug in( it works on a friends laptop), updating my computer software, trying both of the physical ports on my laptop.
It is super frustrating trying to figure this out, and I am at a loss. I have tried searching youtube and google with little success so far. 
I appreciate the help everybody!

**Comment by poundy** [comment]
There were no real issues discussed of late on Hottop so assume it's something you are doing. 
Can you tell us history of you and Artisan and the hottop? Often this is as simple as a new user of all of these things trying to figure out how they all should connect.
You should also use Device Manager to confirm ports. My preference is to have it open and visible without the device connected, then plug it in and confirm what new device pops up. That way I know for sure that the device is what I am expecting and not just a random thing that you already had connected.

**Reply by poundy** [reply]
also exactly what did you mean when you said "works" when you mentioned "getting a new cord to plug in( it works on a friends laptop)" ? Did the cord and artisan and the hottop all work on the friends laptop?

**Reply by 3PY44** [reply]
My buddy and I bought the Hottop. We both have Artisan on our respective laptops for when we are roasting. My Surface Pro9 will not read the Hottop being plugged in, the software boots up and runs, it just won't find the roaster. Using the same cord and the same roaster, my buddy can use Artisan flawlessly. 
I watched multiple videos including the tutorial by Roast rebels on how to connect to a Hottop, I just can't get mine to work. It may be a computer setting somewhere completely apart from Artisan. I just know I have tried everything i can think of.

**Reply by poundy** [reply]
ok excellent, thanks for clearing this up. That is actually a great position to be in. 
First you should take a copy of your buddy's settings file from their computer, and then use them on yours - that way you have a working config that should have no issues except the port. Then, open Device Manager, and look for COM and LPT ports section, without the roaster connected - it might not be visible, which means there's no other serial devices connected. Then, connect the cable from roaster to PC (and turn it on if necessary). Device Manager will likely refresh the display as it deals with the newly detected device, and now you can see the COM port section and see what serial port you need to select, eg COM4.  Then, open Artisan, go into the Config -> Ports menu option and change only the COM port to the one that appeared in Device Manager.  From there, you should be able to test and see  if that has helped.

**Reply by 3PY44** [reply]
Okay, I followed the steps and found where my computer says the ports. But I still cannot seem to get the computer (or roaster) to recognize when they are plugged in. 
![Screenshot 2023-07-15 162518](https://github.com/artisan-roaster-scope/artisan/assets/138802851/45f70bc6-05c0-4685-850a-cc1ae19e1231)
Should the COM ports not say bluetooth??

**Reply by poundy** [reply]
They should say "bluetooth" if that's what you connect. If you are connecting a physical cable between the Surface Pro and the Roaster, then I doubt that you should see Bluetooth mentioned. But you have the perfect test case, what does your friend's laptop show?? Yours should show exactly the same if you're using exactly the same roaster, cable, and connection method.

**Reply by poundy** [reply]
the fact your computer does not recognise a new port when you attach the roaster shows that there's some physical problem with the connection. Whether that's an incompatibility with the SP9's case design and the cable you're using, or some other electrical issue, is unclear (from afar) but on a personal note here, I have for many years used my Surface Book and Surface Book 2 devices connected to my roaster control systems without issue. I did have an issue where one USB device (an external Ethernet adapter) would not be recognised when I connected it to the USB port, and that was "solved" by using a short USB male to female extension lead, and then connecting the device to that. For some reason the electrical connection wasn't made correctly when using just the device. So for you, I would suggest you try using either a USB extension lead, or even a USB hub. You should check the extension and/or hub works with a known working device that works fine when you connect it to the SP9's USB port, and it should still work when connected via the extension/hub. Then, try the roaster, and with any luck you should see the same device as your friend sees in Device Manager
